.circleci/config.yml

```diff
@@ -0,0 +1,333 @@
+      - persist_to_workspace:
+          root: ReactAndroid/src/androidTest/assets/
+          paths:
+            - AndroidTestBundle.js
+
+  test-android:
+    <<: *defaults
+    docker:
+      - image: circleci/android:api-23-alpha
+    environment:
+      - TERM: "dumb"
+      - ADB_INSTALL_TIMEOUT: 10
+      - GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx512m -XX:+HeapDumpOnOutOfMemoryError"'
+      - ANDROID_NDK: '~/ndk/android-ndk-r10e'
+    steps:
+      - checkout
+# The JavaScript Bundle is built as part of the build-js-bundle workflow.
+      - attach_workspace:
+          at: ReactAndroid/src/androidTest/assets/
+      - run:
+          name: Check for JavaScript Bundle
+          command: |
+            if [[ ! -e ReactAndroid/src/androidTest/assets/AndroidTestBundle.js ]]; then
+              echo "JavaScript bundle missing, verify build-js-bundle step"; exit 1;
+            else
+              echo "JavaScript bundle found.";
+            fi
+# CircleCI does not support interpolating env variables in the environment: step above.
+# https://circleci.com/docs/2.0/env-vars/#interpolating-environment-variables-to-set-other-environment-variables
+      - run:
+          name: Configure PATH
+          command: |
+            echo 'export PATH=$ANDROID_NDK:$PATH' >> $BASH_ENV
+            echo 'export PATH=~/react-native/gradle-2.9/bin:~/buck/bin:$PATH' >> $BASH_ENV
+            source $BASH_ENV
+# Configure dependencies
+      - run:
+          name: Install Build Dependencies
+          command: |
+            sudo apt-get update -y
+            sudo apt-get install ant autoconf automake g++ gcc libqt5widgets5 lib32z1 lib32stdc++6 libtool make maven pkg-config python-dev python3-dev qml-module-qtquick-controls qtdeclarative5-dev -y
+      - run:
+          name: Install Android SDKs and Create Android Virtual Device
+          command: source scripts/circle-ci-android-setup.sh && getAndroidSDK
+# Starting emulator in advance as it takes some time to boot.
+      - run:
+          name: Launch Android Virtual Device
+          command: emulator64-arm -avd testAVD -no-audio -no-window -no-boot-anim -gpu off
+          background: true
+# Continue configuring dependencies while AVD boots.
+      - restore-cache: *restore-cache-ndk
+      - run:
+          name: Install Android NDK
+          command: |
+            if [[ ! -e ~/ndk ]]; then
+              mkdir ~/ndk
+              cd ~/ndk
+              curl -o ndk.zip --silent https://dl.google.com/android/repository/android-ndk-r10e-linux-x86.zip
+              curl -o ndk_64.zip --silent https://dl.google.com/android/repository/android-ndk-r10e-linux-x86_64.zip
+              unzip -o ndk.zip
+              unzip -o ndk_64.zip
+              rm ndk.zip
+              rm ndk_64.zip
+            fi
+      - save-cache: *save-cache-ndk
```

Comment:
@ide any ideas how we can only update the cache if `~/ndk` didn't exist? From what I can tell, saving a cache can only be done by specifying a `save-cache` key as a step, so this can't be easily moved into the step above.

Reply:
Actually, updating the cache if ~/ndk didn't exist is not what we want. Leaving this line in should be OK if Circle skips updating the cache if the key already exists.